### ISSUE TITLE
Is it possible to use keras to update specific weight matrix in a matrix function?

### ISSUE THREAD

**Issue by stifflerhe** [fchollet/keras#2211]
My problem is this：I want to use a abnormal layer in the last second layer , and the last layer is  softmax. This abnomal layer has two input matrix X and Y, and the output of it  is exp(XPY), in this case, P is the weight matrix I need to update, I want to know if there are any possibility  to  use keras to do this?
